s3fs/core.py

```diff
@@ -478,6 +480,39 @@ def info(self, path, version_id=None):
             except ParamValidationError as e:
                 raise ValueError('Failed to head path %r: %s' % (path, e))
         return super().info(path)
+    
+    def checksum(self, path, refresh=False):
+        """
+        Unique value for current version of file
+
+        If the checksum is the same from one moment to another, the contents
+        are guaranteed to be the same. If the checksum changes, the contents
+        *might* have changed.
+
+        Parameters
+        ----------
+        path : string/bytes
+            file or path to get checksum for
+        refresh : bool (=False)
+            if False, look in local cache for file details first
+        
+        """
+
+        listing = self.ls(path, detail=True, refresh=refresh)
```

Comment:
Why not `info`? I would have thought that this method works for only a single file, and is probably not useful for a directory (because a directory's contents can have changed without the directory itself having changed, right?)

Reply:
Right. I thought it would be tricky to modify `info` to support bypassing the cache, but on a second look it's probably not. I'll try to fix this.

Reply:
OK, changed this to use `info` instead, and removed test case for directories (since that is unspecified behavior anyway).